src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy

```diff
@@ -42,6 +43,66 @@ class ArtifactoryPermissionsUpdater {
      */
     private static final boolean DEVELOPMENT = Boolean.getBoolean('development')
 
+    /**
+     * Loads all teams from the teams/ folder.
+     * Always returns non null.
+     */
+    private static Map<String, Set<TeamDefinition>>  loadTeams() {
+        Yaml yaml = new Yaml(new Constructor(TeamDefinition.class))
+        File teamsDir = new File('teams/')
+
+        Set<TeamDefinition> teams = [] as Set
+
+        teamsDir.eachFile { teamFile ->
+            try {
+                TeamDefinition newTeam = yaml.loadAs(new FileReader(teamFile), TeamDefinition.class)
+                teams.add(newTeam)
+
+                String expectedName = "team-${newTeam.name}.yml"
```

Comment:
makes more sense to me since you're in the team folder
```suggestion
                String expectedName = "${newTeam.name}.yml"
```

Reply:
yeah, I thought that we may want to add different files in that folder. But for now we can simplify, let me push a fix for this just now.

Reply:
done